modules/apps/content-dashboard/content-dashboard-test/src/testIntegration/java/com/liferay/content/dashboard/web/internal/portlet/action/test/GetContentDashboardItemsXlsMVCResourceCommandTest.java

```diff
@@ -0,0 +1,285 @@
+import com.liferay.portal.kernel.test.portlet.MockLiferayResourceRequest;
+import com.liferay.portal.kernel.test.portlet.MockLiferayResourceResponse;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.GroupTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.FileUtil;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.portal.model.impl.PortletAppImpl;
+import com.liferay.portal.model.impl.PortletImpl;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+
+import java.util.Date;
+import java.util.Iterator;
+
+import org.apache.poi.hssf.usermodel.HSSFWorkbook;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellType;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.mock.web.MockHttpServletRequest;
+
+/**
+ * @author Yurena Cabrera
+ */
+@RunWith(Arquillian.class)
+public class GetContentDashboardItemsXlsMVCResourceCommandTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup(
+			TestPropsValues.getCompanyId(), TestPropsValues.getUserId(), 0,
+			"MyScope");
+	}
+
+	@Test
+	public void testServeResource() throws Exception {
+		_addFileEntry();
+
+		ByteArrayOutputStream generatedByteArrayOutputStream = _serveResource(
+			FileEntry.class.getName(), _group.getGroupId());
+
+		byte[] expectedFile = FileUtil.getBytes(getClass(), "expected.xlsx");
+
+		ByteArrayOutputStream expectedExcelByteArrayOutputStream =
+			new ByteArrayOutputStream();
+
+		expectedExcelByteArrayOutputStream.write(expectedFile);
+
+		_assertEqualsToExpectedFileCellByCell(
+			generatedByteArrayOutputStream, expectedFile);
+	}
+
+	private void _addFileEntry() throws Exception {
+		Date date = new Date(150000);
+
+		DLAppLocalServiceUtil.addFileEntry(
+			"Site", TestPropsValues.getUserId(), _group.getGroupId(),
+			DLFolderConstants.DEFAULT_PARENT_FOLDER_ID, "FileNameXls.pdf",
```

Comment:
The test is failing because this name does not match the one inside the xlsx file (fileNameCsv.pdf).